File: C++/Search in Rotated Sorted Array.cpp
class Solution {
  public:
    int search(vector<int>& arr, int key) {
        for(int i =0;i < arr.size();i++)
        {
            if(arr[i]==key)
            {
                return i;
            }
        }
        return -1;
    }
};

File: JAVA/Search in Rotated Sorted Array.java
class Solution {
    int search(int[] arr, int key) {
        int start = 0;
        int end = arr.length - 1;
        while( start <= end){
            int mid = (start + end)/2;
            if(arr[mid] == key) return mid; 
            if(arr[start] <= arr[mid])     
                if(arr[start] <= key && key <= arr[mid])    end = mid;  
                else    start = mid + 1;                                 
            else                           
                if(arr[mid] <= key && key <= arr[end])  start = mid;     
                else    end = mid-1;                                     
        }
        return -1;
    }
}
